Nowsecure/index.ts

```diff
@@ -1,135 +1,149 @@
-
-if (url) {
-  java.arg("--url");
-  java.arg(url);
+  return tl.tool(toolPath)
 }
 
-if (waitMinutes) {
-  java.arg("--wait");
-  java.arg(waitMinutes);
-} else {
-  java.arg("--wait");
-  java.arg("0");
+type Inputs = {
+  // required params
+  filepath: string;
+  group: string;
+  token: string;
+  // optional params
+  artifact_dir?: string;
+  api_host?: string;
+  ui_host?: string;
+  log_level?: string;
+  analysisType?: string;
+  minimum_score?: string;
+  polling_duration_minutes?: string;
 }
 
-if (showStatusMessages) {
-  java.arg("--show-status-messages");
-  java.arg(showStatusMessages);
-} else {
-  java.arg("--show-status-messages");
-  java.arg("true");
+function getInputs(): Inputs {
+  const inputs = {
+    filepath: tl.getPathInputRequired("binary_file", true),
+    group: tl.getInput("group", true),
+    token: tl.getInput("token", true),
+    // optional params
+    artifact_dir: tl.getInput("artifact_dir", false),
+    api_host: tl.getInput("api_host", false),
+    ui_host: tl.getInput("ui_host", false),
+    log_level: tl.getInput("log_level", false),
+    analysisType: tl.getInput("analysis_type", false),
+    minimum_score: tl.getInput("minimum_score", false),
+    polling_duration_minutes: tl.getInput("polling_duration_minutes", false)
+  }
+
+  if (inputs.analysisType === 'static') {
+    inputs.polling_duration_minutes = inputs.polling_duration_minutes || '30'
+  } else {
+    inputs.polling_duration_minutes = inputs.polling_duration_minutes || '60'
+  }
+
+  return inputs
 }
 
-if (username) {
-  java.arg("--username");
-  java.arg(username);
-}
-if (password) {
-  java.arg("--password");
-  java.arg(password);
-}
-if (scoreThreshold) {
-  java.arg("--score");
-  java.arg(scoreThreshold);
-} else {
-  java.arg("--score");
-  java.arg("0");
-}
+async function run() {
+  const task = JSON.parse(fs.readFileSync(path.join(__dirname, "task.json")).toString());
+  const version = `${task.version.Major}.${task.version.Minor}.${task.version.Patch}`
+
+  const inputs = getInputs()
+
+  tl.mkdirP(inputs.artifact_dir)
+
+  const ns = getTool()
+    .arg("run")
+    .arg("file")
+    .arg(inputs.filepath)
+    .line(`--group-ref ${inputs.group}`)
+    .line(`--token ${inputs.token}`)
+    .line(`--output ${path.join(inputs.artifact_dir, "assessment.json")}`)
+    .line(`--api-host ${inputs.api_host}`)
+    .line(`--ui-host ${inputs.ui_host}`)
+    .line(`--log-level ${inputs.log_level}`)
+    .line(`--analysis-type ${inputs.analysisType}`)
+    .line(`--minimum-score ${inputs.minimum_score}`)
+    .line(`--poll-for-minutes ${inputs.polling_duration_minutes}`)
+    .line(`--ci-environment azure-${version}`)
+
+  ns.on("stdout", function(data: Buffer) {
+    console.log(data.toString());
+  });
+
+  let exitCode: number
+  exitCode = await ns.execAsync()
 
-if (process.env.SYSTEM_DEBUG) {
-  java.arg("--debug");
+  if (exitCode != 0) {
+    const errmsg = "NowSecure extension failed with nonzero exitcode"
```

Comment:
I think the exit code gets lost here unless it's in the console lines

```suggestion
    const errmsg = `NowSecure extension exited with code ${exitCode}`
```